src/renderers/shared/fiber/__tests__/ReactIncremental-test.js

```diff
@@ -281,6 +281,76 @@ describe('ReactIncremental', function() {
     // Flush the rest to make sure that the bailout didn't block this work.
     ReactNoop.flush();
     expect(ops).toEqual(['Middle']);
+  });
+
+  it('can resume work in a bailed subtree within one pass', function() {
+
+    var ops = [];
+
+    function Bar(props) {
+      ops.push('Bar');
+      return <div>{props.children}</div>;
+    }
+
+    function Tester() {
+      // This component is just here to ensure that the bail out is
+      // in fact in effect in the expected place for this test.
+      ops.push('Tester');
+      return <div />;
+    }
+
+    function Middle(props) {
+      ops.push('Middle');
+      return <span>{props.children}</span>;
+    }
+
+    var middleContent = (
+      <aaa>
+        <Tester />
+        <bbb hidden={true}>
+          <ccc>
+            <Middle>Hi</Middle>
+          </ccc>
+        </bbb>
+      </aaa>
+    );
+
+    function Foo(props) {
+      ops.push('Foo');
+      return (
+        <div hidden={props.text === 'bar'}>
+          <Bar>{props.text}</Bar>
+          {middleContent}
+          <Bar>{props.text}</Bar>
+        </div>
+      );
+    }
+
+    // Init
+    ReactNoop.render(<Foo text="foo" />);
+    ReactNoop.flushLowPri(52);
```

Comment:
I don't know if it is multiple of 5 or 6. I just randomly try some numbers until I get the effect I want. :)